Toggl.Core.Tests/UI/ViewModels/CalendarViewModelTests.cs

```diff
@@ -113,16 +150,366 @@ public void EmitsNewDateWhenCurrentlyVisiblePageChanges(int pageIndex, string ex
+                var date = new DateTime(year, month, day);
+                var tappedWeekViewDayViewModel = new CalendarWeeklyViewDayViewModel(date, false, true);
+
+                viewModel.SelectDayFromWeekView.Execute(tappedWeekViewDayViewModel);
                 TestScheduler.Start();
 
                 observer.Values().Should().BeEquivalentTo(expectedResults);
             }
         }
 
+        public sealed class TheCurrentlyShownDateProperty : CalendarViewModelTest
+        {
+            private static readonly DateTimeOffset now = new DateTimeOffset(2020, 1, 2, 3, 4, 5, TimeSpan.Zero).ToLocalTime().Date;
+
+            public TheCurrentlyShownDateProperty()
+            {
+                TimeService.CurrentDateTime.Returns(now);
+            }
+
+            [Theory, LogIfTooSlow]
+            [InlineData(4, 5, 2020)]
+            [InlineData(30, 9, 1995)]
+            public void EmitsNewDateWhenAnItemInWeekViewIsSelected(int day, int month, int year)
+            {
+                var selectedDate = new DateTime(year, month, day);
+                var selectedDayViewModel = new CalendarWeeklyViewDayViewModel(selectedDate, false, true);
+                var preferences = Substitute.For<IThreadSafePreferences>();
+                preferences.DateFormat.Returns(DateFormat.ValidDateFormats[0]);
+                DataSource.Preferences.Current.Returns(Observable.Return(preferences));
+                var expectedResults = new[]
+                {
+                    new DateTime(2020, 1, 2),
+                    selectedDate
+                };
+                var observer = TestScheduler.CreateObserver<DateTime>();
+                var viewModel = CreateViewModel();
+                viewModel.CurrentlyShownDate.Subscribe(observer);
+
+                viewModel.SelectDayFromWeekView.Execute(selectedDayViewModel);
+                TestScheduler.Start();
+
+                observer.Values().Should().BeEquivalentTo(expectedResults);
+            }
+
+            [Fact, LogIfTooSlow]
+            public void StartsWithTheCurrentDate()
+            {
+                var preferences = Substitute.For<IThreadSafePreferences>();
+                preferences.DateFormat.Returns(DateFormat.ValidDateFormats[0]);
+                DataSource.Preferences.Current.Returns(Observable.Return(preferences));
+                var expectedResults = new[]
+                {
+                    now.Date
+                };
+                var observer = TestScheduler.CreateObserver<DateTime>();
+                var viewModel = CreateViewModel();
+                viewModel.CurrentlyShownDate.Subscribe(observer);
+
+                TestScheduler.Start();
+
+                observer.Values().Should().BeEquivalentTo(expectedResults);
+            }
+        }
+
+        public sealed class TheWeekViewDaysProperty : CalendarViewModelTest
+        {
+            private DateTimeOffset now = new DateTimeOffset(2017, 4, 3, 1, 2, 3, TimeSpan.Zero);
```

Comment:
```suggestion
            private DateTimeOffset now = new DateTimeOffset(2017, 4, 3, 1, 2, 3, TimeSpan.Zero).ToLocalTime();
```

This fixes this section of tests for me.